games/xworld3d/tasks/XWorld3DNavTargetSide.py

```diff
@@ -27,57 +28,141 @@ def __init__(self, env):
         super(XWorld3DNavTargetSide, self).__init__(env)
 
     def idle(self):
-        dt = 1.5 # eight connected neighborhood
         goals = self._get_goals()
-        ## because agent loc is continuous, we have to rely on its init position for path search
-        init_agent_loc = self.env.init_agent_loc
+        agent, _, _ = self._get_agent()
 
-        goals = [g for g in goals if \
-                 [e for e in self._get_surrounding_empty_grids(distance_threshold=dt, refer=g) if \
-                  self._reachable(init_agent_loc, e)]]
+        assert len(goals) >= 2, "This task must have at least two goals"
```

Comment:
From the example you provided "Please put the apple to the left of you", it seems at least one goal (rather than two) is enough?

Reply:
That example is outdated.

Reply:
We could either update the example or remove it. Otherwise, it could be mis-leading.